examples/MomentoApplication/Program.cs

```diff
@@ -17,25 +17,50 @@ class Program
         async static Task Main(string[] args)
         {
             using SimpleCacheClient client = new SimpleCacheClient(MOMENTO_AUTH_TOKEN, DEFAULT_TTL_SECONDS);
-            try
-            {
-                client.CreateCache(CACHE_NAME);
+
+            Console.WriteLine("Creating cache with an illegal empty name");
+            CreateCacheResponse emptyCreateResp = client.CreateCache("");
+            if (emptyCreateResp is CreateCacheResponse.Error error) {
+                Console.WriteLine($"Exception error code: {error.Exception.ErrorCode}");
+                Console.WriteLine($"Exception msg: {error.Exception.Message}");
             }
-            catch (AlreadyExistsException)
-            {
-                Console.WriteLine($"Cache with name {CACHE_NAME} already exists.\n");
+
+            Console.WriteLine($"Creating cache {CACHE_NAME}");
+            CreateCacheResponse createResp = client.CreateCache(CACHE_NAME);
+            if (createResp is CreateCacheResponse.Error errorResp2) {
+                if (errorResp2.Exception.ErrorCode == MomentoErrorCode.ALREADY_EXISTS_ERROR) {
+                    Console.WriteLine($"Cache with name {CACHE_NAME} already exists.");
+                } else {
+                    Console.WriteLine($"Create error message: {errorResp2.Exception.Message}");
+                }
             }
-            Console.WriteLine("Listing caches:");
+
+            Console.WriteLine("Listing caches");
             String token = null;
-            do
-            {
-                ListCachesResponse resp = client.ListCaches(token);
-                foreach (CacheInfo cacheInfo in resp.Caches)
-                {
-                    Console.WriteLine(cacheInfo.Name);
+            do {
```

Comment:
@kvcache - pinging you because you're probably the most familiar with C# idioms out of all of our devs.  WDYT about this?  looks like we have to provide a second variable name in order for the smart cast to work.

Reply:
I do not like arbitr abbrevia in variab nam.

Aside from that, I'm not broadly experienced with real modern c# best practices. I know Java has a thing like this too as does rust - seems like a plausible thing to promote but I haven't solicited reviews from a hundred professionals or anything like that. It seems nice to me. I like rebinding the name.

Reply:
yeah we can nitpick about var name choices later, for now i'm just looking for feedback on the API.  I take it you like the rebinding pattern here better than what we had originally proposed where you had to check the enum and then call `.asHit()` etc.  Correct?

Reply:
@gautamomento curious if you have any input on this too.

Reply:
Very much yes I prefer the rebind versus enum check + our own reimplementation and rename of a single-purpose typecast!

I'm seeing green lines in here and it's an "example." I think it's fair to ask whether we want to invent aliases for words like "response" as those new inventions are being proposed here. I read this on my phone and it was unclear where you were drawing attention to (a `do {`?) so I scanned around. Also, the expression in question I think you're looking for feedback on, `if (listResp is ListCachesResponse.Success listSuccessResp)`, is unnecessarily troublesome to parse for meaning as a consequence of abbreviation. Whole word(s) would make this read more naturally. I don't consider this a nit.

Reply:
duly noted.

Reply:
I prefer the rebinding pattern over enum checks, I find it much easier to read and handy to write.